core/creator/src/main/java/io/quarkus/creator/phase/nativeimage/NativeImagePhase.java

```diff
@@ -506,6 +514,44 @@ private boolean isThisGraalVMRCObsolete() {
         return false;
     }
 
+    private static String getLinuxID(String option) {
+        Process process = null;
+        BufferedReader reader = null;
+
+        try {
+            StringBuilder responseBuilder = new StringBuilder();
+            String line;
+
+            ProcessBuilder idPB = new ProcessBuilder().command("id", option);
+            idPB.redirectError(new File("/dev/null"));
+
+            try {
+                process = idPB.start();
+                process.getOutputStream().close();
```

Comment:
This close could theoretically fail.  So, it might be simpler to redirect the input from `/dev/null` as well, so you don't have to handle it at all.

Reply:
Have resolved with refactor